python/GafferTest/MetadataTest.py

```diff
@@ -723,6 +723,43 @@ def __init__( self, name = "MetadataTestNodeD" ) :
 		self.assertEqual( Gaffer.Metadata.value( n["a"], "test" ), "exact" )
 		self.assertEqual( Gaffer.Metadata.value( n["b"], "test" ), "wildcard" )
 
+	def testPreferSpecificWildcards( self ) :
```

Comment:
I've started work on improving the metadata matching as I suggested above. Seems like it's going to work, although I'm going to have to add `...` syntax like the PathMatchers have because it's needed in GafferCortexUI (lots of registrations for potentially nested parameters). Just wanted to mention it so we don't duplicate effort.